src/utils.ts

```diff
@@ -97,7 +105,93 @@ export const getTracerInfo = (): { name: string; version: string } => {
   return { name, version };
 };
 
+const xRayFieldKeyValuePattern = /([^=;]+)=([^;]*)/g;
+
+export const splitXrayTraceIdToFields = (
+  awsXAmznTraceId: string,
+  maxFields: number = 100
+): { [key: string]: string } => {
+  const result: { [key: string]: string | null } = {};
+  let match: RegExpExecArray | null;
+  let iterations = 0;
+
+  // Resetting the regex lastIndex to ensure it works correctly for different inputs
+  xRayFieldKeyValuePattern.lastIndex = 0;
+
+  // Iterate over each match of the pattern in the string with a limit to avoid infinite loop
+  while (
+    (match = xRayFieldKeyValuePattern.exec(awsXAmznTraceId)) !== null &&
+    iterations < maxFields
+  ) {
+    const key = match[1].trim();
+    const value = match[2].trim();
+
+    // Ensure key is valid and not empty, assign null if value is not present
+    if (key) {
+      result[key] = value ? value : null;
+    }
+
+    // Increment iteration counter
+    iterations++;
+  }
+
+  // If maxIterations is reached, we can log a warning or handle it as needed
+  if (iterations >= maxFields) {
+    console.warn(`Reached maximum iteration limit of ${maxFields} while parsing X-Ray trace ID`);
+  }
+
+  return result;
+};
+
+export const getNewFormatTraceId = (awsXAmznTraceId: string): xRayTraceIdFields => {
+  const fields = splitXrayTraceIdToFields(awsXAmznTraceId);
+  const root = fields.Root;
+  if (!root) {
+    throw new Error(
+      `X-Ray trace ID is missing the Root field (_X_AMZN_TRACE_ID=${awsXAmznTraceId})`
+    );
+  }
+  const rootValueSplit = root.split('-');
+  if (rootValueSplit.length < 3) {
+    throw new Error(
+      `X-Ray trace ID Root field is not in the expected format (Root=${fields.Root}, _X_AMZN_TRACE_ID=${awsXAmznTraceId})`
+    );
+  }
+  const transactionId = fields.Root.split('-')[2];
+  // Note: we might not need to generate a Parent field if it's not present,
+  // but for now we'll keep it as is to minimize changes. The python tracer doesn't generate it FYI.
+  const parent = fields.Parent || getRandomString(16);
+  const sampled = fields.Sampled;
+  const lineage = fields.Lineage;
```

Comment:
There's a way to deconstruct ("unpack"), rename and provide a default value all at once, in case you want to reduce assignments:
```suggestion
  const { Parent: parent = getRandomString(16), Sampled: sampled,  Lineage: lineage } = fields;
```